Warning: "possible misuse of comma operator here [-Werror,-Wcomma]"

This is more of a question than a bug report, as I haven't managed to narrow down my configuration enough to give you a minimal repro.

The question is: is the comma operator in [this code](https://github.com/boost-ext/ut/blob/f27b342920e7e050b7b761d5926ee9ef59a6042e/include/boost/ut.hpp#L210) intentional, and if so, could a static case be added to silence the warning?

In some of my builds, where I treat warnings as errors, I am getting this:

```
/Users/clare/Documents/develop/ApprovalTests/ApprovalTests.cpp.CMakeSamples/ut/include/boost/ut.hpp:210:9: error: possible misuse of comma operator here [-Werror,-Wcomma]
    ++pi, ++si;
        ^
/Users/clare/Documents/develop/ApprovalTests/ApprovalTests.cpp.CMakeSamples/ut/include/boost/ut.hpp:210:5: note: cast expression to void to silence warning
    ++pi, ++si;
    ^~~~
    static_cast<void>( )
1 error generated.
```



## Specifications

  - Version: bba087c70a93abb8f900dac04e84feb7895e9575
  - Platform: macOS
  - Subsystem:

```
/opt/local/bin/clang-mp-9.0 --version
clang version 9.0.1 
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /opt/local/libexec/llvm-9.0/bin
```

--- a/example/fatal.cpp
+++ b/example/fatal.cpp
@@ -14,8 +14,6 @@ int main() {
   using namespace boost::ut::literals;
   using boost::ut::fatal;
 
-  static constexpr auto value = 42;  // Change to std::nullopt
-
   "fatal"_test = [] {
     using namespace boost::ut::operators;
     using boost::ut::expect;

--- a/include/boost/ut.hpp
+++ b/include/boost/ut.hpp
@@ -207,7 +207,8 @@ template <class TPattern, class TStr>
     } else if (pattern[pi] != str[si]) {
       return {};
     }
-    ++pi, ++si;
+    ++pi;
+    ++si;
   }
 
   if (si < str.size() or pi < std::size(pattern)) {

--- a/test/ut/ut.cpp
+++ b/test/ut/ut.cpp
@@ -12,6 +12,7 @@
 #include <array>
 #include <cstdlib>
 #include <map>
+#include <numeric>
 #include <sstream>
 #include <string>
 #include <string_view>
@@ -1571,4 +1572,76 @@ int main() {
     test_assert('\n' == std::any_cast<char>(test_cfg.log_calls[0]));
     test_assert("fatal"sv == std::any_cast<const char*>(test_cfg.log_calls[1]));
   }
+
+#if not defined(_MSC_VER)
+  {
+    using namespace ut;
+    auto& test_cfg = ut::cfg<ut::override>;
+    test_cfg = fake_cfg{};
+
+    class calculator {
+     public:
+      auto enter(const int& value) -> void { values_.push_back(value); }
+      auto add() -> void {
+        result_ = std::accumulate(std::cbegin(values_), std::cend(values_), 0);
+      }
+      auto sub() -> void {
+        result_ = std::accumulate(std::cbegin(values_) + 1, std::cend(values_),
+                                  values_.front(), std::minus{});
+      }
+      auto get() const -> int { return result_; }
+
+     private:
+      std::vector<int> values_{};
+      int result_{};
+    };
+
+    bdd::gherkin::steps steps = [](auto& steps) {
+      steps.feature("Calculator") = [&] {
+        steps.scenario("*") = [&] {
+          steps.given("I have calculator") = [&] {
+            calculator calc{};
+            steps.when("I enter {value}") = [&](int value) {
+              calc.enter(value);
+            };
+            steps.when("I press add") = [&] { calc.add(); };
+            steps.when("I press sub") = [&] { calc.sub(); };
+            steps.then("I expect {value}") = [&](int result) {
+              expect(that % calc.get() == result);
+            };
+          };
+        };
+      };
+    };
+
+    // clang-format off
+  "Calculator"_test = steps |
+    R"(
+      Feature: Calculator
+
+        Scenario: Addition
+          Given I have calculator
+           When I enter 40
+           When I enter 2
+           When I press add
+           Then I expect 42
+
+        Scenario: Subtraction
+          Given I have calculator
+           When I enter 4
+           When I enter 2
+           When I press sub
+           Then I expect 2
+    )";
+    // clang-format on
+
+    test_assert(2 == std::size(test_cfg.assertion_calls));
+
+    test_assert("42 == 42" == test_cfg.assertion_calls[0].expr);
+    test_assert(test_cfg.assertion_calls[0].result);
+
+    test_assert("2 == 2" == test_cfg.assertion_calls[1].expr);
+    test_assert(test_cfg.assertion_calls[1].result);
+  }
+#endif
 }

